analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -35,21 +35,20 @@ public override bool ShouldExecute()
         return walker.Result;
     }
 
-    private sealed class SyntaxKindWalker : SafeVisualBasicSyntaxWalker
+    internal sealed class SyntaxKindWalker : SafeVisualBasicSyntaxWalker
     {
         public bool Result { get; private set; }
 
-        public override void Visit(SyntaxNode node)
-        {
-            if (!Result)
-            {
-                Result = node.IsAnyKind(
-                    SyntaxKind.AwaitExpression,
-                    SyntaxKind.ForEachStatement,
-                    SyntaxKind.InvocationExpression,    // For array access arr(42)
-                    SyntaxKind.SimpleMemberAccessExpression);
-                base.Visit(node);
-            }
-        }
+        public override void VisitAwaitExpression(AwaitExpressionSyntax node) =>
+            Result = true;
+
+        public override void VisitForEachBlock(ForEachBlockSyntax node) =>
+            Result = true;
+
+        public override void VisitInvocationExpression(InvocationExpressionSyntax node) =>
+            Result = true; // For array access arr(42)
+
+        public override void VisitMemberAccessExpression(MemberAccessExpressionSyntax node) =>
+            Result = true; // SimpleMemberAccess and DictionaryAccess
```

Comment:
There is also `dict!SomeKey` 
https://learn.microsoft.com/en-us/dotnet/visual-basic/reference/language-specification/expressions#dictionary-member-access-expressions

Reply:
I didn't know this exists 🤯

Reply:
We should have a rule not to use it :D

Reply:
I think it dates back to VBA / MS Access. It was used in the visual forms editor extensively (`Forms!MyForm!tb_UserName` instead of `Forms("MyForm").Controls("tb_UserName")`), or to access fields in a recordset (`rst!PersonId` instead of `rst.Fields("PersonId")`).

https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/program-structure/special-characters-in-code#exclamation-point--operator